src/storage/src/hummock/store/version.rs

```diff
@@ -235,3 +260,369 @@ impl HummockReadVersion {
         self.staging.sst.clear();
     }
 }
+
+pub struct HummockReadSnapshot(
+    pub Vec<ImmutableMemtable>,
+    pub Vec<SstableInfo>,
+    pub CommittedVersion,
+);
+
+impl HummockReadSnapshot {
+    pub fn build_for_batch(
+        epoch: HummockEpoch, // for check
+        table_id: TableId,
+        key_range: &(Bound<Vec<u8>>, Bound<Vec<u8>>),
+        read_version_vec: Vec<Arc<RwLock<HummockReadVersion>>>,
+    ) -> StorageResult<Self> {
+        assert!(!read_version_vec.is_empty());
+        let read_version_guard_vec = read_version_vec
+            .iter()
+            .map(|read_version| read_version.read())
+            .collect_vec();
+        let mut imm_vec = Vec::default();
+        let mut sst_vec = Vec::default();
+        let mut lastst_committed_version =
+            read_version_guard_vec.get(0).unwrap().committed().clone();
+        let mut max_mce = 0;
+
+        // to get max_mce with lock_guard to avoid loosing committed_data since the read_version
+        // update is asynchronous
+        for read_version_guard in &read_version_guard_vec {
```

Comment:
`read_version_guard_vec.iter().map(...).max_by_key()` may make the code cleaner.